### PR TITLE
Server fixes

### PR DESCRIPTION
**Pull request by andsnpl**
This is a set of related changes that puts the GET /api/houstory endpoint back in working order. The core problem was that the POST from the UI saved the user *id*, but when searching, we only have access to the username.

### CODE REVIEW

File: houstory/router.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ router.get('/', jsonParser, (req, res) => {
   //from MyLibrary/library/router.js
   let userId = req.user.id;
   return Houstory.find({userId})
-  .then(posts.map(post.serialize()));
```
Comment (andsnpl) [inline houstory/router.js]: This way tries to evaluate the method calls using `posts` and `post` immediately, instead of chaining them after the `find`.

**Comment by andsnpl** [comment]
Adding a couple more refactoring commits to clean things up...

**Comment by ITDallasDBW** [comment]
I think this is integrating Paul's 'fixes' into FunctionalityAttempt1. The original *.1 has been duped and is named *0.5

**Comment by ITDallasDBW** [comment]
Damn. Pretty sure I did that wrong.

**Pull request merged**